mne/filter.py

```diff
@@ -349,7 +341,10 @@ def _filter(x, Fs, freq, gain, filter_length='10s', picks=None, n_jobs=1,
                           '%0.1fdB. Increase filter_length for higher '
                           'attenuation.' % (att_freq, att_db))
 
-        x = _overlap_add_filter(x, H, zero_phase=True, picks=picks,
+        # reconstruct filter, this time with appropriate gain for fwd-bkwd
+        gain = np.sqrt(gain)
```

Comment:
The gain correction is now implemented here. FWIW it won't do much since our gains are all `0.` or `1.` for our public-facing functions, but it's here for generality.